data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/model/sourcecoordination/SourceCoordinationStoreConfig.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.parser.model.sourcecoordination;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+/**
+ * The interface to be implemented to support different database store configurations
+ * for {@link org.opensearch.dataprepper.model.source.SourceCoordinator} implementations
+ * @since 2.2
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.DEDUCTION)
+@JsonSubTypes({ @JsonSubTypes.Type(DynamoDBSourceCoordinationStoreConfig.class)})
+public interface SourceCoordinationStoreConfig {
```

Comment:
This needs to move to a shared library to make this pluggable outside of data-prepper-core.

Perhaps we should consider having a new library for extending core functionality. Another possible target for this would be extending the metric providers. But, I'm also fine with continuing to use data-prepper-api.

Reply:
I think it would look this way:

```
data-prepper-api/
  SourceCoordinationStoreConfig

data-prepper-core/
  SourceCoordinationConfig

data-prepper-plugins/
  dynamodb-source-coordinator/
    DynamoDBSourceCoordinationStoreConfig
    DynamoStoreSettings
```

Reply:
Correction - I don't think we need the `SourceCoordiationStoreConfig`. Once this uses the plugin interface you probably won't need that class at all.